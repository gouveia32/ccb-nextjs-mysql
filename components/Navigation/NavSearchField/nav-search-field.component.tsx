import React, { useState } from "react";
import { IconButton, TextField } from "@material-ui/core";
import {
  NavSearchFieldButton,
  NavSearchFieldComponent,
} from "./nav-search-field.styles";
import SearchIcon from "@material-ui/icons/Search";
import ClearIcon from "@material-ui/icons/Clear";

import Link from "next/link";
import { PageLinks } from "../../../lib/Links";

export interface NavSearchFieldProps {
  onSearch: (query: string) => void;
  value: string;
}

const NavSearchField: React.FC<NavSearchFieldProps> = ({
  onSearch,
  value,
}: NavSearchFieldProps) => {
  const [showOnMobile, setShowOnMobile] = useState(false);

  return (
    <>
      <NavSearchFieldButton>
        <IconButton onClick={() => setShowOnMobile((prevState) => !prevState)}>
          <SearchIcon />
        </IconButton>
      </NavSearchFieldButton>
      <NavSearchFieldComponent
        showOnMobile={showOnMobile}
        onBlur={() => setShowOnMobile(false)}
      >
        <TextField
          onChange={(event) => onSearch(event.target.value)}
          size={"small"}
          fullWidth={false}
          className="w-90 search-input"
          placeholder={"Procure por notas"}
          value={value}
          InputProps={{
            startAdornment: <SearchIcon fontSize={"small"} className="me-2" />,
            endAdornment: value.length > 0 && (
              <IconButton size={"small"} onClick={() => onSearch("")}>
                <ClearIcon />
              </IconButton>
            ),
              disableUnderline: true
          }}
        />
      </NavSearchFieldComponent>
      
    </>
  );
};

export default NavSearchField;
