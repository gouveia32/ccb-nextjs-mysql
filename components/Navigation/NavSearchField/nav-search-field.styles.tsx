import styled from "styled-components";
import { device } from "../../../resources/styles/utils/media-query-utils";

interface NavSearchFieldComponentInterface {
  showOnMobile?: boolean;
}

export const NavSearchFieldComponent = styled.div<
  NavSearchFieldComponentInterface
>`
  width: 600px;
  margin-left: 5rem;
  padding: 0.5rem 0.3rem 0.2rem 0.8rem;
  background-color: #ededed;
  border-radius: 10px;

  transition: background 100ms ease-in, width 100ms ease-out, all 300ms;

  &:hover,
  &:focus-within {
    background: rgba(255, 255, 255, 1);
    border: 1px solid transparent;
    box-shadow: 0 1px 1px 0 rgba(65, 69, 73, 0.3),
      0 1px 3px 1px rgba(65, 69, 73, 0.15);
  }

  @media only screen and ${device.tablet} {
    width: 300px;
    margin-left: 1rem;
  }

  @media only screen and ${device.mobileL} {
    display: ${(props) => (props.showOnMobile ? "block" : "none")};
    position: fixed;
    margin: 0;
    width: 200px;
    left: 0.5rem;
    top: 0.5rem;
  }
`;

export const NavSearchFieldButton = styled.div`
  display: none;

  @media only screen and ${device.mobileL} {
    width: 100%;
    margin-right: 0.1rem;
    display: flex;
    justify-content: flex-end;
    align-items: center;

    svg {
      color: #fff;
    }
  }
`;
