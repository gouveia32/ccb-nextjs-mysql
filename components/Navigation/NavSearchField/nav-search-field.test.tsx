import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import NavSearchField, {
  NavSearchFieldProps,
} from "./nav-search-field.component";
import { TextField } from "@material-ui/core";

describe("NavSearchField component", () => {
  let wrapper: ShallowWrapper;
  let mockOnSearch: any;

  beforeEach(() => {
    mockOnSearch = jest.fn();

    const mockProps: NavSearchFieldProps = {
      value: "",
      onSearch: mockOnSearch,
    };

    wrapper = shallow(<NavSearchField {...mockProps} />);
  });

  it("should render NavSearchField component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onSearch when TextField si changed", () => {
    wrapper.find(TextField).simulate("change", { target: { value: "" } });
    expect(mockOnSearch).toBeCalled();
  });
});
