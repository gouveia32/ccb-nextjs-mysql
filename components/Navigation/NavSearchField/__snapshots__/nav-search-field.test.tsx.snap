// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSearchField component should render NavSearchField component 1`] = `ShallowWrapper {}`;
